Add: clarify config relationship; fix headings. (#4478)
circleci/circleci-docs: jekyll/_cci2/concepts.md

* _Follow_ any project in your organization to gain access to its pipelines and to subscribe to [email notifications]({{
site.baseurl }}/2.0/notifications/) for the project's status.

![header]({{ site.baseurl }}/assets/img/docs/CircleCI-2.0-setup-project-circle101_cloud.png)

## Configuration

> CircleCI believes in *configuration as code*. Your entire continuous integration/deployment process is orchestrated through a single file called `config.yml`.  The `config.yml` file is located in a folder called `.circleci` at the root of your project. CircleCI uses the YAML syntax for config, see the [Writing YAML]({{ site.baseurl }}/2.0/writing-yaml/) document for basics.

Reviewer: ```suggestion
CircleCI believes in *configuration as code*. Your entire continuous integration and deployment process is orchestrated through a single file called `config.yml`.  The `config.yml` file is located in a folder called `.circleci` at the root of your project. CircleCI uses the YAML syntax for config, see the [Writing YAML]({{ site.baseurl }}/2.0/writing-yaml/) document for basics.
```

```bash
├── .circleci
│   ├── config.yml
├── README

> └── all-your-projects-other-files-and-folders

Reviewer: ```suggestion
└── all-other-project-files-and-folders
```

```

> `config.yml` is a powerful YAML file that defines the entire pipeline for your project. For a full overview of the various keys that can be used see the [Configuration Reference]({{ site.baseurl }}/2.0/configuration-reference/). 

Reviewer: ```suggestion
`config.yml` is a powerful YAML file that defines the entire pipeline for your project. For a full overview of the various keys that are used, see the [Configuration Reference]({{ site.baseurl }}/2.0/configuration-reference/). 
```

> While your CircleCI configuration can be adapted to fit many different needs, the following terms, sorted in order of granularity and dependence, describe the anatomy of most common CircleCI projects:

Reviewer: ```suggestion
Your CircleCI configuration can be adapted to fit many different needs of your project. The following terms, sorted in order of granularity and dependence, describe the components of most common CircleCI projects:
```
